tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/typebynode/TypeByAccessExprTest.java

```diff
@@ -91,9 +91,11 @@ public void visit(IndexedExpressionNode indexedExpressionNode) {
                 if (indexedExpressionNode.keyExpression().size() == 1) {
                     assertType(indexedExpressionNode, model, STRING);
                 } else {
-                    Optional<TypeSymbol> type = assertType(indexedExpressionNode, model, TYPE_REFERENCE);
-                    assertEquals(((TypeReferenceTypeSymbol) type.get()).name(), "Person");
-                    assertEquals(((TypeReferenceTypeSymbol) type.get()).typeDescriptor().typeKind(), RECORD);
+                    Optional<TypeSymbol> type = assertType(indexedExpressionNode, model, UNION);
+                    List<TypeSymbol> members = ((UnionTypeSymbol) type.get()).memberTypeDescriptors();
+                    assertEquals(members.get(0).typeKind(), TYPE_REFERENCE);
+                    assertEquals(((TypeReferenceTypeSymbol) members.get(0)).name(), "Person");
+                    assertEquals(((TypeReferenceTypeSymbol) members.get(0)).typeDescriptor().typeKind(), RECORD);
```

Comment:
Do we need to assert the nil member also here?

Looping in @pubudu91 too.

Reply:
Added an assertion in 3b81168